lib/portlayer/network/context.go

```diff
@@ -933,10 +933,14 @@ func (c *Context) RemoveIDFromScopes(op trace.Operation, id string) ([]*Endpoint
 		return nil, nil // not bound
 	}
 
+	// take a copy of the array we'll be modifying as we iterate
+	tmpEndpoints := con.endpoints
```

Comment:
You are not making a copy of a content if it is a slice. That copy for slices only includes slice metadata.

Reply:
Thanks for the sanity check - should know better than to make a PR from hurried placeholder code but wanted it up as reference in case someone was able to pick it up while I was in the air. Appreciate the weekend review!